From 0678dc534dd1f42427793d2aa3bfd49cfb4db8a3 Mon Sep 17 00:00:00 2001
From: Guo Ren <ren_guo@c-sky.com>
Date: Wed, 9 Oct 2019 17:30:55 +0800
Subject: [PATCH] riscv: Add RISCV_ISA_CUSTOM for gcc

RISC-V could support custom ISA in spec.

Reference riscv-spec: Non-Standard Extension Names

Signed-off-by: Guo Ren <ren_guo@c-sky.com>
---
 arch/riscv/Kconfig  | 11 +++++++++++
 arch/riscv/Makefile |  1 +
 2 files changed, 12 insertions(+)

diff --git a/arch/riscv/Kconfig b/arch/riscv/Kconfig
index ee84d02..5daa8dd 100644
--- a/arch/riscv/Kconfig
+++ b/arch/riscv/Kconfig
@@ -208,6 +208,17 @@ config RISCV_ISA_C
 config RISCV_ISA_A
 	def_bool y
 
+config RISCV_ISA_THEAD
+	bool "T-HEAD Enhanced Instructions"
+
+
+
+
+
+
+
+
+
 menu "supported PMU type"
 	depends on PERF_EVENTS
 
diff --git a/arch/riscv/Makefile b/arch/riscv/Makefile
index c6342e6..55e6c58 100644
--- a/arch/riscv/Makefile
+++ b/arch/riscv/Makefile
@@ -44,6 +44,7 @@ riscv-march-$(CONFIG_ARCH_RV64I)	:= rv64im
 riscv-march-$(CONFIG_RISCV_ISA_A)	:= $(riscv-march-y)a
 riscv-march-$(CONFIG_FPU)		:= $(riscv-march-y)fd
 riscv-march-$(CONFIG_RISCV_ISA_C)	:= $(riscv-march-y)c
+riscv-march-$(CONFIG_RISCV_ISA_THEAD)	:= $(riscv-march-y)xthead
 KBUILD_CFLAGS += -march=$(subst fd,,$(riscv-march-y))
 KBUILD_AFLAGS += -march=$(riscv-march-y)
 
-- 
2.7.4

